core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -577,28 +610,32 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
             // conn.close()
           }
         } while (!locked)
-        (conn, new TableIdentifier(table, Some(schemaName)))
+        Some(conn, new TableIdentifier(table, Some(schemaName)))
       case _ =>
         logDebug(s"Taking lock in " +
-            s" ${Thread.currentThread().getId} ")
+            s" ${Thread.currentThread().getId} and " +
+          s"app ${sqlContext.sparkContext.appName}")
         val regionLock = PartitionedRegion.getRegionLock(SnappySession.WRITE_LOCK_PREFIX + table,
           GemFireCacheImpl.getExisting)
-        regionLock.lock()
-        regionLock
+        regionLock.lock(serializedWriteLockTimeOut * 1000)
```

Comment:
Timeout is in seconds? I think in the property description more clarity can be given in the description.